merge dev into master

mightyBroccoli opened the pull request:
+ added multi host support
* added support for password extraction from prosody.cfg.lua

closes: #6 closes: #9

mightyBroccoli committed:
* read mysql credentials from prosody.cfg.lua
ref.: #9

mightyBroccoli committed:
+ added alpha multi host support
ref.: #6

mightyBroccoli committed:
- removed unnecessary cat love
- removed unnecessary echo
* some shellcheck fixes

imolein commented on .user.conf.example:
```diff
@@ -25,8 +25,8 @@ enable_mam_clearing=false
-prosody_db_user="prosody"
-prosody_db_password="super_secret-password1337"
+prosody_db_user=$(sed -n -e '/username/ s/.*\= *//p' /etc/prosody/prosody.cfg.lua | sed 's/[";]//g')
+prosody_db_password=$(sed -n -e '/password/ s/.*\= *//p' /etc/prosody/prosody.cfg.lua | sed 's/[";]//g')
```
It's simple and it should work like this (don't tested it yet), but I think
* the `prosody.cfg.lua` path should be a variable and not hard coded
* I don't know if there are another matches for `username` or `password` in the cfg, but to prevent a mismatch, you should search for `sql = {` first or something like this

mightyBroccoli replied:
hmm the sql credentials shouldn't be somewhere else but it is possible to make if configurable sure. I searched though all config options in the doc and there are no other possible uses for `username` and `password` possible. But it is totally possible. 
I will fix that.

mightyBroccoli committed:
+ introduced prosody_config_location as variable for the global config
* removed possible malfunction in retrieving the username/password from config

imolein commented:
Nice!

mightyBroccoli committed:
* updated exit code definitions
+ added skriptconfig version check
+ added config test safety catch

mightyBroccoli reviewed (commented):
As there are new variables introduced in this merge. I though about adding a config version check. This will prevent users from accidentally deleting stuff that was not meant to be removed.

mightyBroccoli committed:
* make the config test work again

mightyBroccoli committed:
* fixed logic error

mightyBroccoli committed:
* suppress the output of the filename after using wc

mightyBroccoli committed:
+ added ignore_file backup
* correct execption on \\n with printf
+ implemented account removal with telnet interface
- removed prepared_list and bash execution
